### PR TITLE
[flutter_tools] ensure build fails if asset files are missing

### LINKED ISSUE #53756: Library asset path creation failed: FileSystemException: Creation failed
## Steps to Reproduce

1. Flutter imported in native android app as a module
2. Make some changes in native app
3. Every 10th build fails with given error, then have to clear Flutter to make it work

**Actual results:** 

```
> Task :flutter:compileFlutterBuildDebug
Target debug_android_application failed: FileSystemException: Creation failed, path = '/xyz/flutter_module/.android/Flutter/build/intermediates/flutter/debug/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out/flutter_assets/.android/Flutter/build/intermediates/library_assets/debug/out' (OS Error: File name too long, errno = 63)
build failed.

> Task :flutter:compileFlutterBuildDebug FAILED

FAILURE: Build failed with an exception.

* Where:
Script '/x/flutter-sdk/packages/flutter_tools/gradle/flutter.gradle' line: 780

* What went wrong:
Execution failed for task ':flutter:compileFlutterBuildDebug'.
> Process 'command '/x/flutter-sdk/bin/flutter'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 36s
1 actionable task: 1 executed
```

Seems that there's something wrong with the generated path?

The module was created with `flutter create flutter_module`

```

[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76, locale en-PL)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.6)
[✓] Connected device (2 available)
```

</details>

### LINKED ISSUE #53946: "Apply Code Changes" for Android doesn't work because AssetManifest.json is changed every build
There's a nice feature in Android Studio to apply code changes without restarting Android app. I have Flutter embedded as a module and I use is a FlutterActivity.

Every time I make single line change in Android related XML, apply code changes can't do its work because of this error:

```
04/03 21:47:53: Applying code changes to 'app' on 8.0 720p.
Changes were not applied.
Modifying resources requires an activity restart.
Resource 'assets/flutter_assets/AssetManifest.json' was modified.
Apply changes and restart activity
```

This happens only for Android projects with Flutter. 

```
[✓] Flutter (Channel stable, v1.12.13+hotfix.9, on Mac OS X 10.15.3 19D76, locale en-PL)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.6)
[✓] Connected device (1 available)
```

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Ensure that the build fails if the asset bundle cannot be built. Also ensure that stderr is output from the flutter assemble task, otherwise `printError` messages do not appear in gradle output without verbose.

Fixes https://github.com/flutter/flutter/issues/53946
Fixes https://github.com/flutter/flutter/issues/53756 (?)

### CODE REVIEW

File: packages/flutter_tools/gradle/flutter.gradle [hunk 1]
```diff
@@ -881,6 +881,7 @@ abstract class BaseFlutterTask extends DefaultTask {
             ruleNames = targetPlatformValues.collect { "android_aot_bundle_${buildMode}_$it" }
         }
         project.exec {
+            logging.captureStandardError LogLevel.ERROR
```
Comment (blasten) [inline packages/flutter_tools/gradle/flutter.gradle:884]: should we update or add a new integration test?

**Review by blasten — APPROVED** [review]
LGTM

File: packages/flutter_tools/gradle/flutter.gradle (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/gradle/flutter.gradle:884]: Yes we should, easy to miss this one.

File: packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart [hunk 2]
```diff
@@ -97,6 +97,26 @@ flutter:
...
+name: example
+
+flutter:
+  assets:
+    - assets/foo/bar2.png
+
+''');
+
+    expect(() async => await const CopyAssets().build(environment),
+      throwsA(isA<Exception>()));
```
Comment (blasten) [inline packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart:114]: qq: is it worth checking the exception message as well?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart:114]: The exception is pretty nothingburger, the meaningful message is over stderr from checks like https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/flutter_manifest.dart#L206

**Review by dnfield — APPROVED** [review]

File: packages/flutter_tools/gradle/flutter.gradle (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/gradle/flutter.gradle:884]: Added a test, as verifies the output message

File: dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart [hunk 3]
```diff
@@ -190,6 +190,19 @@ Future<void> main() async {
               result);
       });
 
+      await runProjectTest((FlutterProject project) async {
```
Comment (jonahwilliams) [inline dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart:193]: FYI @blasten

Comment (blasten) [reply dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart:193]: 🎉

File: dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart [hunk 4]
```diff
@@ -190,6 +190,19 @@ Future<void> main() async {
...
+        section('gradlew assembleDebug forwards stderr');
+        await project.introducePubspecError();
+                final ProcessResult result =
```
Comment (zanderso) [inline dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart:196]: weird indentation.

**Review by zanderso — COMMENTED** [review]
dbc

**Pull request merged**